src/creature/creature.cpp

```diff
@@ -23,6 +23,55 @@ Creature::Creature(int hp,
 	  m_ability_scores{strength, dexterity, constitution, intelligence, wisdom, charisma},
 	  m_creature_type(creature_type){};
 
+void Creature::set_hp(int hp)
+{
+	if (hp >= 0)
+	{
+		m_hp = hp;
+	}
+	else
+	{
+		std::cerr << "The HP stat of Creature cannot be negative!\n";
+	}
+}
+
+void Creature::set_ac(int ac)
+{
+	if (ac >= 0)
+	{
+		m_ac = ac;
+	}
+	else
+	{
+		std::cerr << "The AC stat of Creature cannot be negative!\n";
+	}
+}
+
+void Creature::set_speed(int speed)
+{
+	if (speed >= 0)
+	{
+		m_speed = speed;
+	}
+	else
+	{
+		std::cerr << "The SPEED stat of Creature cannot be negative!\n";
+	}
+}
+
+void Creature::set_ability_score(int ability_score, int value)
+{
+	if (ability_score > -1 && ability_score < 6 && value > -1)
+	{
+		m_ability_scores[ability_score] = value;
+	}
+	else
+	{
+		std::cerr << "The " << m_ability_scores[ability_score]
```

Comment:
I believe this will just output: "The [some number] stat of Creature cannot be negative!\n"

It isn't necessary to have the precise name of the stat. Just make it say: "The ability scores of Creature cannot be negative!\n"